client/components/spinner/index.jsx

```diff
@@ -66,12 +69,23 @@ export default class Spinner extends PureComponent {
 				isSVGCSSAnimationSupported: isSVGCSSAnimationSupported
 			} );
 		}
+
+		if ( this.props.delay ) {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+
+			this.timer = setTimeout( () => this.setState( { show: true } ), this.props.delay );
```

Comment:
This triggers a js error if the component is not mounted when this fires. At the very least this needs to be cleared on unmount, but lets save these spinner improvements for another PR